firefox-ios/firefox-ios-tests/Tests/ClientTests/Telemetry/TermsOfServiceTelemetryTests.swift

```diff
@@ -7,67 +7,162 @@ import XCTest
+        XCTAssert(savedMetric === event, "Received \(savedMetric) instead of \(event)")
     }
 
     func testRecordTermsOfServiceAcceptButtonTappedThenGleanIsCalled() throws {
+        let subject = createSubject()
         let acceptedDate = Date()
-        subject?.termsOfServiceAcceptButtonTapped(acceptedDate: acceptedDate)
-        try testEventMetricRecordingSuccess(metric: GleanMetrics.Onboarding.termsOfServiceAccepted)
-
-        // Test that ToU accepted event is recorded with onboarding surface
-        try testEventMetricRecordingSuccess(metric: GleanMetrics.TermsOfUse.accepted)
-        let acceptedEventValue = try XCTUnwrap(GleanMetrics.TermsOfUse.accepted.testGetValue())
-        XCTAssertEqual(acceptedEventValue[0].extra?["surface"], "onboarding")
-        let expectedVersionString = String(TermsOfUseTelemetry().termsOfUseVersion)
-        XCTAssertEqual(acceptedEventValue[0].extra?["tou_version"], expectedVersionString)
-
-        // Test that ToU version and date metrics are also recorded for consistency
-        let versionValue = try XCTUnwrap(GleanMetrics.UserTermsOfUse.versionAccepted.testGetValue())
-        XCTAssertEqual(versionValue, TermsOfUseTelemetry().termsOfUseVersion)
-        let dateValue = try XCTUnwrap(GleanMetrics.UserTermsOfUse.dateAccepted.testGetValue())
-        XCTAssertTrue(Calendar.current.isDate(dateValue, inSameDayAs: acceptedDate))
+        let expectedVersion = "4"
+        let expectedSurface = "onboarding"
+
+        let onboardingEvent = GleanMetrics.Onboarding.termsOfServiceAccepted
+        let touAcceptedEvent = GleanMetrics.TermsOfUse.accepted
+        let versionMetric = GleanMetrics.UserTermsOfUse.versionAccepted
+        let dateMetric = GleanMetrics.UserTermsOfUse.dateAccepted
+        typealias AcceptedExtra = GleanMetrics.TermsOfUse.AcceptedExtra
+
+        subject.termsOfServiceAcceptButtonTapped(acceptedDate: acceptedDate)
+
+        XCTAssertEqual(mockGleanWrapper.recordEventNoExtraCalled, 1)
+        XCTAssertEqual(mockGleanWrapper.recordEventCalled, 1)
+        XCTAssertEqual(mockGleanWrapper.recordQuantityCalled, 1)
+        XCTAssertEqual(mockGleanWrapper.recordDatetimeCalled, 1)
+
+        let savedOnboardingEvent = try XCTUnwrap(
+            mockGleanWrapper.savedEvents[0] as? EventMetricType<NoExtras>
+        )
+        XCTAssert(savedOnboardingEvent === onboardingEvent, "Received \(savedOnboardingEvent) instead of \(onboardingEvent)")
+
+        let savedTouEvent = try XCTUnwrap(
+            mockGleanWrapper.savedEvents[1] as? EventMetricType<AcceptedExtra>
+        )
+        XCTAssert(savedTouEvent === touAcceptedEvent, "Received \(savedTouEvent) instead of \(touAcceptedEvent)")
+
+        let savedExtras = try XCTUnwrap(
+            mockGleanWrapper.savedExtras.first as? AcceptedExtra
+        )
+        XCTAssertEqual(savedExtras.surface, expectedSurface)
+        XCTAssertEqual(savedExtras.touVersion, expectedVersion)
```

Comment:
nit: it might be easier to view the actual value instead of jumping back to where the value was defined, no strong opinions though.
  
```suggestion
        XCTAssertEqual(savedExtras.surface, "onboarding")
        XCTAssertEqual(savedExtras.touVersion, "4")
```